crates/repository/src/lib.rs

```diff
@@ -364,6 +364,15 @@ impl Repository {
     pub async fn set_location(&self, target: &str, location: &str) -> Result<()> {
         let target_location = self.parameters_root().join(format!("{target}_location"));
         let new_location = Path::new(location);
+        let new_location_path = self.parameters_root().join(location);
+        if !new_location_path.exists() {
+            let mut available_locations = String::new();
+            let location_set = self.list_available_locations().await?;
+            for location in location_set {
+                writeln!(&mut available_locations, "  -{location}")?;
+            }
+            bail!("available locations are:\n{available_locations}but you wrote: {location}");
```

Comment:
IMHO, the error should be first, and then the hints and suggestions to the user. Thus, swap `available_locations` with the missing `location`

Reply:
Done